### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by deniss-mc**
Starting Engagement with audio and video calls functionality

Chat Bubble functionality still needs to be modified after these changes

### CODE REVIEW

File: .idea/compiler.xml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
```
Comment (mustafa-ogun-ozturk) [inline .idea/compiler.xml:1]: unnecessary file

File: .idea/misc.xml [hunk 2]
```diff
@@ -54,4 +54,4 @@
   <component name="ProjectType">
     <option name="id" value="Android" />
```
Comment (mustafa-ogun-ozturk) [inline .idea/misc.xml:55]: unnecessary file

File: widgetssdk/src/main/java/com/glia/widgets/helper/TimeCounter.java [hunk 3]
```diff
@@ -76,14 +76,14 @@ private void createNewTimerTask(int timerDelayMs, int timerIntervalMs) {
             public void run() {
                 Logger.d(TAG, "Timer value: " + value);
                 for (RawTimerStatusListener listener : rawListeners) {
-                    listener.onNewTimerValue(value);
+                    listener.onNewRawTimerValue(value);
                 }
 
                 if (!formattedListeners.isEmpty()) {
-                    String time = Utils.toMmSs(Long.valueOf(TimeUnit.MILLISECONDS.toSeconds(value)).intValue());
+                    String time = Utils.toMmSs(Long.valueOf(value));
```
Comment (mustafa-ogun-ozturk) [inline widgetssdk/src/main/java/com/glia/widgets/helper/TimeCounter.java:83]: valueof call can go to the method as well

File: widgetssdk/src/main/java/com/glia/widgets/head/ChatHeadsController.java [hunk 4]
```diff
@@ -206,6 +228,7 @@ public void onNewOperatorMediaState(OperatorMediaState operatorMediaState) {
         Logger.d(TAG, "new operatorMediaState: " + operatorMediaState);
         handleService();
         emitViewState(chatHeadState.setOperatorMediaState(operatorMediaState));
+
```
Comment (mustafa-ogun-ozturk) [inline widgetssdk/src/main/java/com/glia/widgets/head/ChatHeadsController.java:231]: extra line

File: widgetssdk/src/main/java/com/glia/widgets/chat/ChatView.java [hunk 5]
```diff
@@ -212,20 +215,32 @@ public void startChat(
...
+            controller.initChat(companyName, queueId, contextUrl);
+
+            controller.updatePermissions(
                     Settings.canDrawOverlays(this.getContext()),
                     NotificationManager.areNotificationsEnabled(this.getContext(), NotificationFactory.NOTIFICATION_CALL_CHANNEL_ID),
                     NotificationManager.areNotificationsEnabled(this.getContext(), NotificationFactory.NOTIFICATION_SCREEN_SHARING_CHANNEL_ID)
             );
+
+            if (activity instanceof ChatActivity && savedInstanceState == null) {
+                if (chatHeadsController != null) {
```
Comment (mustafa-ogun-ozturk) [inline widgetssdk/src/main/java/com/glia/widgets/chat/ChatView.java:230]: seems like unnecessary nested if

**Review by mustafa-ogun-ozturk — APPROVED** [review]

File: widgetssdk/src/main/java/com/glia/widgets/chat/ChatView.java (hunk 5 above)
Comment (deniss-mc) [reply widgetssdk/src/main/java/com/glia/widgets/chat/ChatView.java:230]: Will look into it while looking into chat bubble issues

File: .idea/compiler.xml (hunk 1 above)
Comment (deniss-mc) [reply .idea/compiler.xml:1]: gitignore needs updating, will deal with it next week

File: .idea/misc.xml (hunk 2 above)
Comment (deniss-mc) [reply .idea/misc.xml:55]: gitignore needs updating, will deal with it next week

**Pull request merged**